src/agent/onefuzz-agent/src/tasks/fuzz/libfuzzer_fuzz.rs

```diff
@@ -178,34 +171,23 @@ impl LibFuzzerFuzzTask {
     }
 
     async fn init_directories(&self) -> Result<()> {
-        utils::init_dir(&self.config.inputs.path).await?;
-        utils::init_dir(&self.config.crashes.path).await?;
+        self.config.inputs.init().await?;
+        self.config.crashes.init().await?;
         if let Some(readonly_inputs) = &self.config.readonly_inputs {
             for dir in readonly_inputs {
-                utils::init_dir(&dir.path).await?;
-            }
-        }
-        Ok(())
-    }
-
-    async fn sync_all_corpuses(&self) -> Result<()> {
-        utils::sync_remote_dir(&self.config.inputs, utils::SyncOperation::Pull).await?;
-
-        if let Some(readonly_inputs) = &self.config.readonly_inputs {
-            for corpus in readonly_inputs {
-                utils::sync_remote_dir(corpus, utils::SyncOperation::Pull).await?;
+                dir.init().await?;
             }
         }
-
         Ok(())
     }
 
-    async fn resync_all_corpuses(&self) -> Result<()> {
-        loop {
-            time::delay_for(RESYNC_PERIOD).await;
-
-            self.sync_all_corpuses().await?;
+    async fn continuous_sync_inputs(&self) -> Result<()> {
+        let mut dirs = vec![self.config.inputs.clone()];
+        if let Some(inputs) = &self.config.readonly_inputs {
+            let mut inputs = inputs.clone();
+            dirs.append(&mut inputs);
```

Comment:
I'm a little confused by the types here. Why are we appending a mutable reference?